docs/DeveloperGuide.md

```diff
@@ -260,6 +266,87 @@ Note that only the `Category` field is allowed to be specified multiple times, w
 a `ParserException` is thrown. Another noteworthy point is that `Category` that are added are to be case-insensitively unique and can only be up to 
 a specified value in the `UniqueCategoryList` class. Else, a `ParserException` would be thrown.
 
+#### Delete Transaction
+
+##### Overview
+
+The `DeleteCommand` function deletes an existing `Transaction` from `TransactionList` in UniCash.
+
+The activity diagram of deleting a Transaction is as shown below
+
+<img src="images/unicash/DeleteTransactionActivityDiagram.png" width="600" />
+
+The following sequence diagram shows the interaction between different components of UniCash.
+
+<img src="images/unicash/DeleteTransactionSequenceDiagram.png" width="1200" />
+
+The above sequence diagram omits details on the filtering of `TransactionList` and assumes that
+the displayed `TransactionList` is showing all transactions. However, the logic of the `DeleteCommand`
+remains the same for all list deletion.
+
+ℹ️ **Note:** The lifeline for `DeleteTransactionCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML,
+the lifeline reaches the end of diagram.
+
+##### Details
+
+1. The user specifies the transaction to be deleted by stating the integer index of the transaction to be deleted.
```

Comment:
Should we specify that the integer index entered is a first-indexed one, rather than a zero-indexed one?

Reply:
In the `TransactionList` the integer indices are all one-based, so when the user "states .. the index of the transaction", it's implies that the integer index is one-indexed as the user will be referring to the transaction number that they see beside the transaction they are interested in.

Thinking about it, since this applies to every command involving an index, I think when we update next week we can have a similar group of specifications or add to our glossary about what is meant by "Transaction index" vs zero-based index.